src/bin/diff.js

```diff
@@ -2,41 +2,120 @@
 
 'use strict'
 
+var path = require('path')
+var argv = require('yargs')
+  .boolean('verbose')
+  .alias('v', 'verbose')
+  .argv
+
 var size = require('window-size')
 var availableWidth = size.width || /* istanbul ignore next */ 80
 var ui = require('cliui')({width: availableWidth})
 
 var getRuleFinder = require('../lib/rule-finder')
-var difference = require('../lib/array-diff')
+var arrayDifference = require('../lib/array-diff')
+var objectDifference = require('../lib/object-diff')
 var getSortedRules = require('../lib/sort-rules')
 
-var rules = getSortedRules(
-  difference(
-    getRuleFinder(process.argv[2]).getCurrentRules(),
-    getRuleFinder(process.argv[3]).getCurrentRules()
-  )
-)
-
-var outputRules, outputRuleCellMapper
 var outputPadding = ' '
 var outputMaxWidth = 0
 var outputMaxCols = 0
 
+var files = [argv._[0], argv._[1]]
+var collectedRules = getFilesToCompare(files).map(compareConfigs)
+
+var rulesCount = collectedRules.reduce(
+  function getLength(prev, curr) {
+    return prev + (curr && curr.rules ? curr.rules.length : /* istanbul ignore next */ 0)
+  }, 0)
+
+var outputRuleCellMapper
+
 /* istanbul ignore next */
-if (rules.length) {
-  console.log('\n diff rules\n') // eslint-disable-line no-console
-  rules = rules.map(function columnSpecification(rule) {
-    rule = rule + outputPadding
-    outputMaxWidth = Math.max(rule.length, outputMaxWidth)
-    return rule
-  })
-  outputMaxCols = Math.floor(availableWidth / outputMaxWidth)
+if (rulesCount) {
+  console.log('\ndiff rules') // eslint-disable-line no-console
+
+  outputMaxCols = argv.verbose ? 3 : Math.floor(availableWidth / outputMaxWidth)
   outputRuleCellMapper = getOutputRuleCellMapper(Math.floor(availableWidth / outputMaxCols))
-  while (rules.length) {
-    outputRules = rules.splice(0, outputMaxCols).map(outputRuleCellMapper)
-    ui.div.apply(ui, outputRules)
+
+  collectedRules.forEach(function displayConfigs(config) {
+    var rules = config.rules
+
+    if (!rules.length) {
+      return
+    }
+
+    if (argv.verbose) {
+      rules.unshift('', config.base, config.head)
+    } else {
+      console.log( // eslint-disable-line no-console
+        '\nin ' + config.base + ' but not in ' + config.head + ':\n' +
+        rules.length + ' rules found\n'
+      )
+    }
+
+    while (rules.length) {
+      ui.div.apply(
+        ui,
+        rules.splice(0, outputMaxCols).map(outputRuleCellMapper)
+      )
+    }
+    console.log(ui.toString()) // eslint-disable-line no-console
+  })
+}
+
+function getFilesToCompare(allFiles) {
+  var filesToCompare = [allFiles]
```

Comment:
But it's currently required.

* `files` specifies the files to compare
* `filesToCompare` describes which files to compare, and in what order

For non-verbose output, we need to do two comparisons in total, first compare a to b, second compare b to a.